### PR TITLE
ENH Remove TF warnings about lambda functions

### PR DESCRIPTION
**Pull request by janden**

### CODE REVIEW

**Review by edouardoyallon — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     docker:
       # specify the version you desire here
       # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
-      - image: circleci/python:3.6.1
+      - image: circleci/python:3.7.1
```
Comment (MuawizChaudhary) [inline .circleci/config.yml:15]: Do we want to change this to 3.8.1? or 3.9.0?

File: README.md [hunk 2]
```diff
@@ -140,12 +145,15 @@ python setup.py develop
 
 ## GPU acceleration
 
-The available frontends are PyTorch (provided with the backends `torch`, `torch_skcuda`), TensorFlow (with the backend
-`tensorflow`), and NumPy (with the backend `numpy`).
+Certain frontends, `numpy` and `sklearn`, only allow processing on the CPU and are therefore slower. The `torch`, `tensorflow`, and `keras` frontends, however, also support GPU processing, which can significantly accelerate computations. Additionally, the `torch` backend supports an optimized `skcuda` backend which currently provides the fastest performance in computing scattering transforms. In 2D, it may be instantiated using:
```
Comment (MuawizChaudhary) [inline README.md:148]: "The `torch`, `tensorflow`, and `keras` frontends also support GPU processing, which can significantly accelerate computations." ?

File: doc/source/userguide.rst [hunk 3]
```diff
@@ -112,43 +120,80 @@ size:
...
+    S = Scattering2D(J=2, shape=(32, 32))
 
-Additionally, you can choose a frontend via the `frontend=` arguments available in all the `Scattering` constructors. By default, `torch`
-frontend is used.
+Alternatively, the object may be instantiated in a dynamic way using the :class:`kymatio.Scattering2D` object by providing a ``frontend`` argument. This object then transforms itself to the desired frontend. Using this approach, the above example becomes::
 
+    from kymatio import Scattering2D
+    S = Scattering2D(J=2, shape=(32, 32), frontend='numpy')
 
+In Kymatio 0.2, the default frontend is ``torch`` for backwards compatibility reasons, but this change to ``numpy`` in the next version.
```
Comment (MuawizChaudhary) [inline doc/source/userguide.rst:154]: "but this will be changed to ``numpy`` in the next version?"

File: doc/source/userguide.rst [hunk 4]
```diff
@@ -195,8 +259,8 @@ alternative.
 For instance, the default backend of the ``torch`` frontend is the ``torch`` backend,
 implemented exclusively in PyTorch. This is available for 1D, 2D, and 3D. It is also
 compatible with the PyTorch automatic differentiation framework, and runs on
-both CPU and GPU. If one wants additional improved performance on GPU, it is recommended
-to use the ``torch_skcuda`` backend.
+both CPU and GPU. If one wants additional improved performance on GPU, we
+recommended to use the ``torch_skcuda`` backend.
```
Comment (MuawizChaudhary) [inline doc/source/userguide.rst:263]: "we recommend using the torch_skcuda backend"

File: doc/source/userguide.rst [hunk 5]
```diff
@@ -220,6 +284,10 @@ This backend can be specified via::
 
     scattering = Scattering2D(J=2, shape=(32, 32), backend='torch_skcuda')
 
+Each of the other frontends currently only has a single backend, which is the
+default. Work is currently underway, however, to extend some of these frontends
```
Comment (MuawizChaudhary) [inline doc/source/userguide.rst:288]: "Work is currently underway to extend some of these frontends" OR "However, work is currently underway to extend some of these frontends"

**Review by MuawizChaudhary — COMMENTED** [review]
Want me to make these changes?

**Comment by janden** [comment]
@MuawizChaudhary It looks like this PR is inadvertently including some unrelated commits (probably due to a rebase of the dev branch). I will rebase it and push it to update.

Can you turn your comments into an issue (or issues) since they relate to other parts of the code (or make PRs with your suggested changes)?

**Review by MuawizChaudhary — COMMENTED** [review]
Why is 3d not changed here?

**Comment by MuawizChaudhary** [comment]
We should merge this after #573 .

File: kymatio/scattering1d/backend/tensorflow_backend.py [hunk 6]
```diff
@@ -113,6 +113,10 @@ def unpad(x, i0, i1):
     return x[..., i0:i1]
 
 
+_fft1d = lambda x: tf.signal.fft(x, name='fft1d')
```
Comment (eickenberg) [inline kymatio/scattering1d/backend/tensorflow_backend.py:116]: if you have to name the function, why not just use `def`

File: kymatio/scattering2d/backend/tensorflow_backend.py [hunk 7]
```diff
@@ -74,18 +74,18 @@ def __call__(self, x, k):
         out = tf.reduce_mean(y, axis=(1, 3))
         return out
 
-
+_fft2d = lambda x: tf.signal.fft2d(x, name='fft2d')
```
Comment (eickenberg) [inline kymatio/scattering2d/backend/tensorflow_backend.py:77]: I'd prefer `def`

File: .circleci/config.yml (hunk 1 above)
Comment (eickenberg) [reply .circleci/config.yml]: I guess we need to test several of the currently actively used ones

File: README.md (hunk 2 above)
Comment (eickenberg) [reply README.md]: Is there a problem with this sentence or was that a suggestion to fix? It looks like the sentence in the doc is exactly that.

Comment (MuawizChaudhary) [reply README.md]: Suggestion, theres a however in the current document which kinda ruins my reading experience.

**Comment by eickenberg** [comment]
> It looks like this PR is inadvertently including some unrelated commits (probably due to a rebase of the dev branch)

Are these doc changes because of that rebasing issue?

**Comment by janden** [comment]
> Are these doc changes because of that rebasing issue?

Correct. Should be gone now.

**Pull request closed without merging**